api/envoy/config/core/v3/base.proto

```diff
@@ -444,3 +444,12 @@ message ControlPlane {
   // the Envoy is connected to.
   string identifier = 1;
 }
+
+// DNS resolver configuration which includes the underlying dns resolver addresses and options
+message DnsResolver {
+  // A list of dns resolver addresses
+  // Setting this value causes failure if the
+  // ``envoy.restart_features.use_apple_api_for_dns_lookups`` runtime value is true during
+  // server startup. Apple's API only allows overriding DNS resolvers via system settings.
+  repeated Address dns_resolvers = 1;
```

Comment:
nit: Should we just name it just `resolvers` instead of `dns_resolvers` ?

Reply:
updated